services-operator/beamsqlstatementsetoperator.py

```diff
@@ -0,0 +1,282 @@
+        # In this case, we need to signal that stage
+        patch.status[STATE] = States.UNKNOWN.name
+
+def create_ddl_from_beamsqltables(body, beamsqltable, logger):
+    """
+    creates an sql ddl object out of a beamsqltables object
+    Works only with kafka connector
+    column names (fields) are not expected to be escaped with ``. This is inserted explicitly.
+    The value of the fields are supposed to be prepared to pass SQL parsing, e.g.
+    value: STRING # will be translated into `value` STRING, value is an SQL keyword (!)
+    dvalue: AS CAST(`value` AS DOUBLE) # will b translated into `dvalue` AS CAST(`value` AS DOUBLE), `value` here is epected to be masqued
+
+    Parameters
+    ----------
+    body: dict
+        Beamsqlstatementset which is currently processed
+    beamsqltable: dict
+        Beamsqltable object
+    logger: log object
+        Local log object provided from framework
+    """
+    name = beamsqltable.metadata.name
+    ddl = f"CREATE TABLE `{name}` ("
+    for key, value in beamsqltable.spec.get("fields").items():
+        if key == "watermark":
+            insert_key = key
+        else:
+            insert_key = f"`{key}`"
+        ddl += f"{insert_key} {value},"
+    ddl = ddl[:-1] # remove last "," from ddl, enumeration is done
+    ddl += ") WITH ("
+    if beamsqltable.spec.get("connector") != "kafka":
+        message = f"Beamsqltable {name} has not supported connector."
+        kopf.warn(body, reason="invalid CRD", message = message)
+        logger.warn(message)
+        return None
+    ddl += "'connector' = 'kafka'"
+    # loop through the value structure
+    # value.format is mandatory
+    value = beamsqltable.spec.get("value")
+    if not value:
+        message = f"Beamsqltable {name} has no value description."
+        kopf.warn(body, reason="invalid CRD", message = message)
+        logger.warn(message)
+        return None
+    if not value.get("format"):
+        message = f"Beamsqltable {name} has no value.format description."
+        kopf.warn(body, reason="invalid CRD", message = mesage)
+        logger.warn(message)
+    for v_key, v_value in value.items():
+        ddl += f",'{v_key}' = '{v_value}'"
+    # loop through the kafka structure
+    # map all key value pairs to 'key' = 'value',
+    # except properties
+    kafka = beamsqltable.spec.get("kafka")
+    if not kafka:
+        message = f"Beamsqltable {name} has no Kafka connector descriptor."
+        kopf.warn(body, reason="invalid CRD", message = message)
+        logger.warn(message)
+        return None
+    # check mandatory fields in Kafka, topic, bootstrap.server
+    if not kafka.get("topic"):
+        message = f"Beamsqltable {name} has no kafka topic."
+        kopf.warn(body, reason="invalid CRD", message = message)
+        logger.warn(message)
+        return None
+
+    if not kafka.get("properties").get("bootstrap.servers"):
```

Comment:
`.get` is designed for cases, in which you expect not to have the key. Here, it would crash with "None has no get" in this case, therefore it is better to use `kafka["properties"]` and get a `KeyError`if this is ever to happen.